### PR TITLE
refactor: support cose envelope

### PR DESCRIPTION
**Pull request by chloeyin**
## What
Background can be checked in https://github.com/notaryproject/notation-go/pull/131
This is the second PR

### CODE REVIEW

File: registry/repository_test.go [hunk 1]
```diff
@@ -531,6 +546,35 @@ func TestPutSignatureManifest(t *testing.T) {
...
+			},
+			args: args{
+				reference:    validReference,
+				signature:    make([]byte, 0),
+				ctx:          context.Background(),
+				remoteClient: mockRemoteClient{},
+				annotations: map[string]string{
+					artifactspec.AnnotationArtifactCreated: validTimestamp,
+				},
+				signatureMediaType: coseTag,
```
Comment (priteshbandi) [inline registry/repository_test.go:577]: Why is cose test different from jws?

File: plugin/plugin.go [hunk 2]
```diff
@@ -192,13 +192,11 @@ type Signature struct {
 // CriticalAttributes contains all Notary V2 defined critical
 // attributes and their values in the signature envelope
 type CriticalAttributes struct {
-	ContentType                  string                 `json:"contentType"`
-	SigningScheme                string                 `json:"signingScheme"`
-	Expiry                       *time.Time             `json:"expiry,omitempty"`
-	AuthenticSigningTime         *time.Time             `json:"authenticSigningTime,omitempty"`
-	VerificationPlugin           string                 `json:"verificationPlugin,omitempty"`
-	VerificationPluginMinVersion string                 `json:"verificationPluginMinVersion,omitempty"`
```
[the remark this answers is not in the document]
Comment (chloeyin) [reply plugin/plugin.go:200]: https://github.com/notaryproject/notaryproject/blob/main/signature-specification.md#extended-attributes-for-notation-plugins. According to the spec, these two attributes should be in the `ExtendedAttributes` and extracting them when using plugin to verify.
In notation-core-go, https://github.com/notaryproject/notation-core-go/blob/f14510cf21c244182574a71f26b920057a0d5bf3/signature/types.go#L28-L41, these two attributes won't be included. I moved this logic to notation go https://github.com/notaryproject/notation-go/blob/2bcfd343f974938e8eeb76e233bc2526e0c36061/verification/verifier_helpers.go#L319-L364

File: config/config.go [hunk 3]
```diff
@@ -23,6 +23,8 @@ type Config struct {
 	InsecureRegistries       []string                 `json:"insecureRegistries"`
 	CredentialsStore         string                   `json:"credsStore,omitempty"`
 	CredentialHelpers        map[string]string        `json:"credHelpers,omitempty"`
+	// EnvelopeType defines the envelope type for signing
+	EnvelopeType string `json:"envelopeType,omitempty"`
```
[the remark this answers is not in the document]
Comment (chloeyin) [reply config/config.go:27]: This will be used in the notation CLI. @JeyJeyGao Can you clarify this?

File: registry/repository_test.go (hunk 1 above)
Comment (chloeyin) [reply registry/repository_test.go:577]: What do you mean by different? This test is just pushing a signature with type `application/cose`.

File: config/config.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (JeyJeyGao) [reply config/config.go:27]: The `--envelope-type` flag will be added in notation-cli: `notation sign --envelope-type cose xxxx`, and `EnvelopeType` in the config.json file is the default value for this cli flag. If `EnvelopeType` is not set, `jws` will be the default value.

[the remark this answers is not in the document]
Comment (priteshbandi) [reply config/config.go:27]: Can we move this out of current cr and add defaulting notation cli? Basically let notation decide the default behaviour

File: registry/repository_test.go (hunk 1 above)
Comment (priteshbandi) [reply registry/repository_test.go:577]: So in jws test we only have `expectDes` while in cose we have `expectDes`, `expectManifest` and `args`? Am I missing something?

File: plugin/plugin.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (priteshbandi) [reply plugin/plugin.go:200]: got it, we are cleaning  left over from previous CR.

File: registry/repository_test.go (hunk 1 above)
Comment (chloeyin) [reply registry/repository_test.go:577]: image
You can expand the line above, it is actually the same :)

File: config/config.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (JeyJeyGao) [reply config/config.go:27]: According this issue: https://github.com/notaryproject/notation/issues/325, we need to add a parameter in config.json to store this setting. If we remove it, we still need to add it in an other PR.

File: registry/repository_test.go (hunk 1 above)
Comment (priteshbandi) [reply registry/repository_test.go:577]: I remembered expanding the lines but somehow missed it :(

File: config/config.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (priteshbandi) [reply config/config.go:27]: Can we add it in the PR where we actually implement https://github.com/notaryproject/notation/issues/325? This PR is for cose envelope support and lets restrict the scope to cose envelope support.

Also regarding system (global) configuration option for signature format, we haven't finalized the approach. There are some edgecases like in presence of plugin how will default behave? Can user override this default by passing parameter while calling sign command (if yes what will be parameter name?), etc. I would like to iron out these cases before we add this support.

[the remark this answers is not in the document]
Comment (JeyJeyGao) [reply config/config.go:27]: Sure. Let's remove it.

**Review by priteshbandi — APPROVED** [review]
LGTM

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**